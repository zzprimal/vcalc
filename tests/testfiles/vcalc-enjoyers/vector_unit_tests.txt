print(1..16);

// Arithmetic
print(1..16 + 1..16);
print(1..16 - 1..16);
print(1..16 * 1..16);
print(1..16 / 1..16);

// Int vector arithmetic
print(1..16 + 1);
print(1..16 - 2);
print(1..16 * 3);
print(1..16 / 4);

print(1 + 1..16);
print(2 - 1..16);
print(3 * 1..16);
print(4 / 1..16);

// Booleans
print(1..16 == 1..16);
print(1..16 != 1..16);
print(1..16 > 1..16);
print(1..16 < 1..16);

print(1..8 < 2..9);
print(2..9 > 1..8);

print(1..16 < 0..(0 - 2147483647));

// Index
print([i in 0..1 | i][0]);
print(1..((1..16)[15]));
print((1..16)[0]);
vector a = 1..16;
print([i in a | i][0]);
//CHECK_FILE:./vector_unit_tests.out
