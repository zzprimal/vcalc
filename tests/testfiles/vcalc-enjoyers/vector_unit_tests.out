[1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16]
[2 4 6 8 10 12 14 16 18 20 22 24 26 28 30 32]
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
[1 4 9 16 25 36 49 64 81 100 121 144 169 196 225 256]
[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1]
[2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17]
[-1 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14]
[3 6 9 12 15 18 21 24 27 30 33 36 39 42 45 48]
[0 0 0 1 1 1 1 2 2 2 2 3 3 3 3 4]
[2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17]
[1 0 -1 -2 -3 -4 -5 -6 -7 -8 -9 -10 -11 -12 -13 -14]
[3 6 9 12 15 18 21 24 27 30 33 36 39 42 45 48]
[4 2 1 1 0 0 0 0 0 0 0 0 0 0 0 0]
[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1]
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
[1 1 1 1 1 1 1 1]
[1 1 1 1 1 1 1 1]
[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
0
[1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16]
1
1
